text/0008-ingest-gem-content.md

```diff
@@ -0,0 +1,332 @@
+- Feature Name: Ingest Gem File Contents
+- Start Date: 2023-01-18
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+Create a process for the purpose of processing and saving the contents of a gem.
+The process will run each time a gem is uploaded to save information about each file in the gem.
+The saved output will include file size, content type, SHA256 checksum, and relative path within the gem.
+In the interest of simplifying this RFC and the feature work that follows, this RFC focuses on only the ingestion and storage of gem content.
```

Comment:
I think it's worth also include gem metadata, specifically the gem spec file (calling it out because the canonical gem spec is stored in a separate tarball than the rest of the gem content)

Reply:
If I understand correctly, you mean pulling the metadata.gz file out of the gem and hosting that along with the gem's content? That makes sense to me.